app/views/provider/admin/api_docs/show.html.erb

```diff
@@ -12,11 +13,31 @@
   <p class="InfoBox InfoBox--notice"><%= link_to 'Create an access token', provider_admin_user_access_tokens_path %> to authenticate against the Account Management API, the Analytics API and the Billing API.</p>
 <% end %>
 
-<div class='api-docs-wrap'></div>
-<script>
-$(function(){
-   ThreeScale.APIDocs.account_type = 'provider';
-   ThreeScale.APIDocs.host = "<%=j base_url.gsub(%r{/$}, '') %>";
-   ThreeScale.APIDocs.init([]);
-});
-</script>
+<div class='api-docs-wrap' id = "analytics-api-docs"></div>
+<div class='api-docs-wrap' id = "billing-api-docs"></div>
+<div class='api-docs-wrap' id = "service-management-api-docs"></div>
+<div class='api-docs-wrap' id = "account-management-api-docs"></div>
+<div class='api-docs-wrap' id = "policy-registry-api-docs"></div>
```

Comment:
I'd prefer to not create a static list of containers. Different APIs are returned for different users, so there will be always some empty container in the code. Also, if we add a new API in the future, we'll need to add a new line here for it. In order to avoid that, it'd be preferable to create the containers dynamically with javascript, one for each API returned by the services call. That is, remove this entire list and replace it by a single `<div id="api-containers">` and then, in the callback for the successful ajax call (`success`) add a new `<div class='api-docs-wrap' id="{api-name}"></div>` to `#api-containers` for each API in the response.

Reply:
Good Suggestion . Again JS challenge for me

Reply:
Done and pushed